config/initializers/lazy_load_routes.rb

```diff
@@ -0,0 +1,42 @@
+# To lazy load routes:
+# 1) Modify the RoutesReloader to skip normal reload! unless opted-in or Rails server
+# 2) Insert a Rack::Middleware to catch the first request with empty routes, opt-in and reload!
+# 3) Modify the RoutesInspector used by rake routes to detect empty routes, opt-in and reload!
+module RoutesReloaderLazyLoadRoutes
+  def reload!
+    if $force_routes_load || defined?(Rails::Server)
+      puts "loading routes..."
```

Comment:
Is this here to stay, or here just for now?  Does it make sense instead to hook it up to the a `Rails` based logger?

Reply:
nah, that was debugging so I knew when it loaded the routes